src/couch/src/couch_server.erl

```diff
@@ -86,18 +86,23 @@ open(DbName, Options0) ->
         {ok, Db1} = couch_db:incref(Db0),
         couch_db:set_user_ctx(Db1, Ctx);
     _ ->
-        Options = maybe_add_sys_db_callbacks(DbName, Options0),
-        Timeout = couch_util:get_value(timeout, Options, infinity),
-        Create = couch_util:get_value(create_if_missing, Options, false),
-        case gen_server:call(couch_server, {open, DbName, Options}, Timeout) of
-        {ok, Db0} ->
-            {ok, Db1} = couch_db:incref(Db0),
-            couch_db:set_user_ctx(Db1, Ctx);
-        {not_found, no_db_file} when Create ->
-            couch_log:warning("creating missing database: ~s", [DbName]),
-            couch_server:create(DbName, Options);
-        Error ->
-            Error
+        check_dbname(DbName),
```

Comment:
`check_dbname/1` doesn't throw? I think it returns `ok | {error, ...}`

Reply:
Derp. Good call. With the shard locking thing I think we can clean up these two checks more.